### PR TITLE
Enable/Disable sms-based password recovery

### PR DESCRIPTION
**Pull request by piraveena**
This includes the following changes:

- Disable/Enable SMS based password recovery; frontport of https://github.com/wso2-support/identity-governance/pull/235

merge this PR after this https://github.com/wso2-extensions/identity-governance/pull/394

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java [hunk 1]
```diff
@@ -42,10 +42,15 @@
...
 import org.wso2.carbon.identity.recovery.util.Utils;
 
+import org.wso2.carbon.identity.user.functionality.mgt.UserFunctionalityManager;
+import org.wso2.carbon.identity.user.functionality.mgt.exception.UserFunctionalityManagementException;
+import org.wso2.carbon.identity.user.functionality.mgt.model.FunctionalityLockStatus;
 import org.wso2.carbon.registry.core.utils.UUIDGenerator;
 import org.wso2.carbon.user.api.UserStoreException;
 import org.wso2.carbon.user.core.UserStoreManager;
 import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java:52]: Is this import needed?

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java [hunk 2]
```diff
@@ -62,6 +67,10 @@
     private static final String FORWARD_SLASH = "/";
     private static final NotificationChannels[] notificationChannels = {
             NotificationChannels.EMAIL_CHANNEL, NotificationChannels.SMS_CHANNEL};
+    private static final boolean PER_USER_FUNCTIONALITY_LOCKING_ENABLED = Utils.isPerUserFunctionalityLockingEnabled();
+    private static final String NOTIFICATION_BASED_PW_RECOVERY_SMS = "NOTIFICATION_BASED_PW_RECOVERY_SMS";
+    private static final String FUNCTIONALITY_PREFIX = "FUNCTIONALITY_";
+
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java:73]: Remove redundant line

Comment (piraveena) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java:71]: fixed

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java [hunk 3]
```diff
@@ -42,11 +42,15 @@
...
 
+import org.wso2.carbon.identity.user.functionality.mgt.UserFunctionalityManager;
+import org.wso2.carbon.identity.user.functionality.mgt.exception.UserFunctionalityManagementException;
+import org.wso2.carbon.identity.user.functionality.mgt.model.FunctionalityLockStatus;
 import org.wso2.carbon.registry.core.utils.UUIDGenerator;
 import org.wso2.carbon.user.api.UserStoreException;
 import org.wso2.carbon.user.core.UserStoreManager;
 import org.wso2.carbon.user.core.service.RealmService;
 
+
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java:53]: Remove redundant line

**Review by sachiniWettasinghe — APPROVED** [review]

**Pull request merged**